tests/scripts/generate_tls13_compat_tests.py

```diff
@@ -95,18 +102,30 @@ def add_signature_algorithms(self, *signature_algorithms):
         self._sig_algs.extend(
             [sig_alg for sig_alg in signature_algorithms if sig_alg not in self._sig_algs])
 
-    # add_signature_algorithms should not override by sub class
+    # add_named_groups should not override by sub class
     def add_named_groups(self, *named_groups):
         self._named_groups.extend(
             [named_group for named_group in named_groups if named_group not in self._named_groups])
 
+    # add_cert_signature_algorithms should not override by sub class
+    def add_cert_signature_algorithms(self, *signature_algorithms):
+        self._cert_sig_algs.extend(
+            [sig_alg for sig_alg in signature_algorithms if sig_alg not in self._cert_sig_algs])
+
     @abc.abstractmethod
     def pre_checks(self):
         return []
 
     @abc.abstractmethod
     def cmd(self):
-        pass
+        if not self._ciphers:
```

Comment:
That's not what I had in mind. `_ciphers` would be the list of forced ciphers. If there is no forced cipher then we do not force anything. Thus for Mbed TLS client there should be no "force_ciphersuite" in the command, for OpenSSL no "-ciphersuite" and I think for GnuTLS the best would be to have CIPHER-ALL.